tidy3d/plugins/mode/solver.py

```diff
@@ -218,16 +221,56 @@ def solver_em(eps_tensor, mu_tensor, der_mats, num_modes, neff_guess):
         Imaginary part of the effective index, shape (num_modes, ).
     """
 
+    # Determine if ``eps`` and ``mu`` are diagonal or tensorial
     off_diagonals = (np.ones((3, 3)) - np.eye(3)).astype(bool)
     eps_offd = np.abs(eps_tensor[off_diagonals])
     mu_offd = np.abs(mu_tensor[off_diagonals])
+    is_tensorial = False
     if np.any(eps_offd > 1e-6) or np.any(mu_offd > 1e-6):
-        return solver_tensorial(eps_tensor, mu_tensor, der_mats, num_modes, neff_guess)
+        is_tensorial = True
```

Comment:
Shall we also set the 1e-6 here as a global variable at the beginning of the script? Also we can put the sp.eigs tolerance up there too (currently fp_eps).